src/git-directory-helper.ts

```diff
@@ -38,6 +44,7 @@ export async function prepareExistingDirectory(
     }
 
     try {
+      core.startGroup('Removing previously created refs, to avoid conflicts')
```

Comment:
I think we need to keep mark a flag to indicate we have started a group but not ended it so we can error handle correctly on this try-catch

Reply:
Unclosed groups dont really hurt anything. Intentionally did not complicate the code to track.

Also note, can't have nested groups. Starting a new group automatically closes any open group.

Thoughts?

Reply:
Otherwise try/finally may make sense